src/main/java/levi9/UserService/service/impl/UserServiceImpl.java

```
@@ -1,27 +1,40 @@
 package levi9.UserService.service.impl;
 
-import levi9.UserService.dto.CredentialsDto;
-import levi9.UserService.dto.LoggedUserDto;
+import levi9.UserService.dto.UserDto;
 import levi9.UserService.model.User;
+import levi9.UserService.model.enums.UserRole;
 import levi9.UserService.repository.UserRepository;
 import levi9.UserService.service.UserService;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepository;
+    private ModelMapper mapper;
 
     @Autowired
     public UserServiceImpl(UserRepository userRepository) {
         this.userRepository = userRepository;
+        this.mapper = new ModelMapper();
     }
 
     @Override
-    public LoggedUserDto authenticate(CredentialsDto credentials) {
+    public UserDto authenticate(UserDto credentials) {
         User user = userRepository.findByUsername(credentials.getUsername());
-        if(user != null && user.getPassword().equals(credentials.getPassword())){
-            return new LoggedUserDto(null, user.getRole(), user.getUsername());
+        if (user != null && user.getPassword().equals(credentials.getPassword())) {
+            return UserDto.builder().role(user.getRole()).username(user.getUsername()).build();
+        }
+        return null;
+    }
+
+    @Override
+    public UserDto register(UserDto newUser) {
+        if (userRepository.findByUsername(newUser.getUsername()) == null) {
+            User user = mapper.map(newUser, User.class);
+            user.setRole(UserRole.CLIENT);
+            return mapper.map(userRepository.save(user), UserDto.class);
         }
         return null;
```

Service can throw an exception here, instead of controller checking for null.